DOC: Tighten howto-docs guide #16259

bjnath opened the pull request:
PR picks up from #16259. It addresses the two comments made in the earlier PR.

![image](https://user-images.githubusercontent.com/6691888/82928970-f2ec6480-9f50-11ea-9acd-23b4a077881b.png)

bjnath committed:
DOC: Latest for #16259

Attempting to fix bad commit 967a735.

bjnath committed:
DOC: Remove unneeded howto-docs clause

Deleted "and it may seem your input wasn't valued." Not clear, and
not necessary in the context of the sentence that follows.

bjnath commented:
Is anything further standing in the way of merging this PR? Happy to address any remaining feedback.

The [most recent review comment](https://github.com/numpy/numpy/pull/16259#pullrequestreview-417855404) ("Overall, I think this looks good!") suggested the PR might be ready to merge once two points of feedback were addressed.  I've made those changes.

Please let me know if anything from me is still needed.

rossbar reviewed (approved):
LGTM - AFAICT the content is largely consistent between this version and the original and it looks like all of the major points are covered.

On a separate (but related) note: at one point in the discussion, there was talk of perhaps coming up with a companion doc (either a "how-to" or a "tutorial") that explicitly walks users through the process of building docs. I think that's a great idea as the doc building process can be a bit involved, especially for new users.

bjnath committed:
Merge branch 'master' into howto-docs-edit

bjnath committed:
DOC: Rewrite 'hand-written' section in how-to docs PR 6259

Per https://github.com/numpy/numpy/pull/16385/#discussion_r438320504.

Other rewordings throughout.

bjnath committed:
DOC: Rewrite 'hand-written' section in how-to docs PR 6259

Per https://github.com/numpy/numpy/pull/16385/#discussion_r438320504.

Other rewordings throughout.:

bjnath commented:
8192a03 addresses the comments about "handwritten" by @rossbar and @mattip in https://github.com/numpy/numpy/pull/16385/#discussion_r438320504 and has
other revisions throughout.

<details><summary>New page:</summary>

![image](https://user-images.githubusercontent.com/6691888/84387008-52d54300-abc0-11ea-9722-9a6f54bf652e.png)

</details>

bjnath commented on doc/source/dev/howto-docs.rst:
Please see if the wording in 8192a03 is an improvement.

bjnath replied:
6043b0f revises it again to make it more straightforward.

bjnath committed:
DOC: Fix repeated word in how-to docs PR $16259

Removed a repeated "into" in the paragraph beginning,
"NumPy documentation is kept..."

bjnath commented:
2c145be fixes a repeated word.

bjnath committed:
DOC: Dropping forest/trees in PR #16259

Going with a more straightforward wording.

Adding verbs to the 'fixes' and 'new pages' headings.

bjnath commented:
6043b0f has more straightforward wording in the section about generated docs and adds verbs to two headings.

<details><summary>New page:</summary>

![image](https://user-images.githubusercontent.com/6691888/84389555-35a27380-abc4-11ea-8ba9-14985ab3a1c9.png)


</details>

bjnath committed:
DOC: Fixing repeated word in PR #16259

Google reference at bottom says "free" twice.

bjnath commented:
75c5908 fixes another repeated word.

<details><summary>New page:</summary>
![image](https://user-images.githubusercontent.com/6691888/84391168-9a5ecd80-abc6-11ea-963c-66ff4327cd4e.png)

</details>

mattip commented:
@bjnath you do not need to attach an image. If you click on the "Details" link in the "ci/circleci: build artifact" build, you will be directed to the built documentation, and you can paste a link to the page here instead.

Also, you don't need to comment on each commit (unless you want to). Those of us who are subscribed to the PR get a mail each time you commit with the commit message. That is why we request that the commit messages be [formatted](https://13992-908607-gh.circle-artifacts.com/0/doc/build/html/dev/development_workflow.html#writing-the-commit-message) in a specific way: that make filtering the many emails easier.

mattip commented:
Thanks @bjnath

bjnath commented:
@mattip -- Thank you for the suggestions. 

> you don't need to comment on each commit

In other words, the commit message is sufficient and I don't have to return to this page and summarize it?

> commit messages be formatted in a specific way

I'd begun all the commits with DOC and referenced the PR in the first line -- is there something else stylistically I should do when the commit updates a pending PR?

> do not need to attach an image

Great! I'll attach the link as you suggest.


Sorry for the last-minute flurry of changes -- I appreciate your patience.

mattip commented:
> the commit message is sufficient

As a rule of thumb: yes. If the new commit is missed you could ping a reviewer after a day or two (a week for really complicated things).

>I'd begun all the commits with DOC and referenced the PR in the first line -- is there something else stylistically I should do when the commit updates a pending PR?

No, my bad, sorry. I wonder why my filter failed?

Thanks for the contributions.
